jwst/assign_wcs/pointing.py

```diff
@@ -237,3 +237,110 @@ def create_fitswcs(inp, input_frame=None):
+    -------
+    va_corr : astropy.modeling.CompoundModel, astropy.modeling.models.Identity
+        A 2D compound model that corrects DVA. If input model does not contain
+        enough information to compute correction then
+        `astropy.modeling.models.Identity` will be returned.
+
+    """
+    if datamodel is None:
+        try:
+            v2_ref = float(kwargs['v2_ref'])
+            v3_ref = float(kwargs['v3_ref'])
+            va_scale = float(kwargs['va_scale'])
+        except KeyError:
+            raise KeyError(
+                "'v2_ref', 'v3_ref', and 'va_scale' are all required when "
+                "'datamodel' is set to None."
+            )
+        except ValueError:
+            raise TypeError(
+                "'v2_ref', 'v3_ref', and 'va_scale' must be numbers."
+            )
+
+    else:
+        if kwargs:
+            raise ValueError(
+                "'v2_ref', 'v3_ref', and 'va_scale' cannot be provided when "
+                "'datamodel' is not None."
+            )
+        try:
+            va_scale = float(datamodel.meta.velocity_aberration.scale_factor)
+            v2_ref = float(datamodel.meta.wcsinfo.v2_ref)
+            v3_ref = float(datamodel.meta.wcsinfo.v3_ref)
+        except (AttributeError, TypeError):
+            return Identity(2)
+
+    v2_shift = (1 - va_scale) * v2_ref
+    v3_shift = (1 - va_scale) * v3_ref
+
+    #  NOTE: it is assumed that v2, v3 angles are small enough so that
+    #  for expected scale factors the issue of angle wrapping (180 degrees)
+    #  can be neglected.
+    va_corr = (
+        (Scale(va_scale, name='va_scale_v2') & Scale(va_scale, name='va_scale_v3')) |
+        (Shift(v2_shift, name='va_v2_shift') & Shift(v3_shift, name='va_v3_shift'))
+    )
+
+    # INFO: Code below does not assume small angle approximation and performs
+    # complete computations (Euler rotations and tangent plane projection).
+    # This code may be useful if VA code will be used with instruments further
+    # away from the optical axis (velocity is assumed parallel to the optical
+    # axis).
+    #
+    # from astropy.modeling.models import Identity
+    #
+    # s2c = SphericalToCartesian(name='s2c', wrap_lon_at=180)
+    # c2s = CartesianToSpherical(name='c2s', wrap_lon_at=180)
+    #
+    # unit_conv = Scale(1.0 / 3600.0, name='arcsec_to_deg_1D')
+    # unit_conv = unit_conv & unit_conv
+    # unit_conv.name = 'arcsec_to_deg_2D'
+    #
+    # va_scale_corr = (Identity(1, name='I_1D') &
+    #                  Scale(va_scale, name='va_scale_1Dy') &
+    #                  Scale(va_scale, name='va_scale_1Dz'))
+    # va_scale_corr.name = 'va_scale_2D'
+    #
+    # rot = RotationSequence3D(
+    #     [v2_ref / 3600.0, -v3_ref / 3600.0],
+    #     'zy',
+    #     name='det_to_optic_axis'
+    # )
+    #
+    # va_corr = (
+    #     unit_conv | s2c | rot | va_scale_corr |
+    #     rot.inverse | c2s | unit_conv.inverse
+    # )
+
+    va_corr.name = 'VA_Correction'
```

Comment:
The name should be descriptive, and bits are free.  "Spacecraft Velocity Aberration Correction" would be useful.

Reply:
@nden What do you think? I personally think that `VA_Correction` should be OK but if you think a longer string is more desirable - it is easy to change this.

Reply:
Models can be referenced by name so I'd say shorter is better and not having spaces in the name is preferable.